app/src/forms/auth/middleware/apiAccess.js

```diff
@@ -21,6 +22,15 @@ module.exports = async (req, res, next) => {
       } else if (params.formSubmissionId && uuidValidate(params.formSubmissionId)) {
         const result = await submissionService.read(params.formSubmissionId);
         formId = result?.form?.id;
+      } else if (params.id && uuidValidate(params.id)) {
+        // get submissionID from request body
+        const sid = await fileService.read(params.id);
+        //check to see that an associated submissionId exists
+        if (!sid || !sid.formSubmissionId) {
+          throw new Error('Submission ID not found in file storage.');
+        }
+        const result = await submissionService.read(sid.formSubmissionId);
+        formId = result?.form?.id;
```

Comment:
Please add tests for this to `app/tests/unit/forms/auth/middleware/apiAccess.spec.js`.

Reply:
I've added some tests now!